genai-perf/genai_perf/types.py

```diff
@@ -56,6 +56,11 @@
 GpuMetricObjectives: TypeAlias = Dict[ModelName, MetricObjectives]
 PerfMetricObjectives: TypeAlias = Dict[ModelName, MetricObjectives]
 
+###########################################################################
+# Parameters
+###########################################################################
+Parameters: TypeAlias = Dict[str, Any]
```

Comment:
I think the better approach is to define a new (sub)type instead of type alias. This is because type alias treats the alias and the original to be _equivalent_ types, but since `Dict[str, Any]` is pretty generic type used across the codebase and not all `Dict[str, Any]` is `Parameters`, we don't want to treat the two as equals but rather a _derived_ type. 

From the Python doc on typing: https://docs.python.org/3/library/typing.html#newtype
> Note: Recall that the use of a type alias declares two types to be equivalent to one another. Doing `type Alias = Original` will make the static type checker treat `Alias` as being exactly equivalent to `Original` in all cases. This is useful when you want to simplify complex type signatures.
In contrast, `NewType` declares one type to be a subtype of another. Doing `Derived = NewType('Derived', Original)` will make the static type checker treat `Derived` as a subclass of `Original`, which means a value of type `Original` cannot be used in places where a value of type `Derived` is expected. This is useful when you want to prevent logic errors with minimal runtime cost.

```suggestion
Parameters = NewType("Parameters", Dict[str, Any])
```

Similar discussion in SO: https://stackoverflow.com/a/52512686

Reply:
This is a good suggestion, but I'm going to push this off for now as it breaks type checking in all the unit testing (since you can no longer input a str and need to cast to the new type).